### DISCUSSION TITLE
Running evaluation without training

### DISCUSSION THREAD

**Discussion by LSnyd** [NVIDIA/NVFlare#660] [Q&A] [answered]
Hi all, 
my project is based on the cifar example and I am trying to run the cross-site validation process without a prior training.
My target is to test the resulting models on stratified datasets to see if the performance is better on certain image groups. 
I tried to remove the scatter and gather workflow from the config, so that the evaluation starts immediately. However, the run folder always gets deleted beforehand and therefore the evaluation workflow has no models to evaluate. 
Is there a way to run it separately at all? Or is it better to set up a separate evaluation pipeline in my case?

**Comment by holgerroth** [comment]
Thank you for raising this issue. The current cifar-10 example is not set up for running evaluation only. However, you can customize the [model_locator](https://github.com/NVIDIA/NVFlare/blob/c8d51bfb534c02eff874a760c47086c87cbff59f/examples/cifar10/configs/cifar10_fedavg/config/config_fed_server.json#L71) specified in the `CrossSiteModelEval` workflow to fit your needs. The evaluated models could be located anywhere on the server or client. By default, we use the [PTFileModelLocator](https://github.com/NVIDIA/NVFlare/blob/0753aba7ad97e09db474e0d56ce843e4eee8285a/nvflare/app_common/pt/pt_file_model_locator.py).

**Comment by YuanTingHsieh** [comment, marked as answer]
You need to change `cleanup_models` to `false` in your config for `CrossSiteModelEval` workflow.

If you don't change this the models will be deleted (cleaned up).

(For example: https://github.com/NVIDIA/NVFlare/blob/main/examples/cifar10/configs/cifar10_fedavg/config/config_fed_server.json#L74)

I don't think cifar example will delete the run folder.

**Reply by YuanTingHsieh** [reply]
@LSnyd has this issue been solved for you?

**Reply by holgerroth** [reply]
@LSnyd can we mark this discussion as answered?

**Reply by LSnyd** [reply]
Yes this was solved, thank you :)
